tests/Unit/PointwiseFunctions/AnalyticData/WaveEquation/Test_RegularSphericalWaveKerrSchild.cpp

```diff
@@ -0,0 +1,247 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <memory>
+
+#include "DataStructures/DataBox/Prefixes.hpp"  // IWYU pragma: keep
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"  // IWYU pragma: keep
+#include "Evolution/Systems/ScalarWave/System.hpp"      // IWYU pragma: keep
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "PointwiseFunctions/AnalyticData/WaveEquation/RegularSphericalWaveKerrSchild.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "PointwiseFunctions/MathFunctions/Gaussian.hpp"
+#include "PointwiseFunctions/MathFunctions/MathFunction.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/TestCreation.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+struct RegularSphWaveKerr {
+  using type = CurvedScalarWave::AnalyticData::RegularSphericalWaveKerrSchild;
+  static constexpr OptionString help{
+      "A regular spherical wave in Kerr spacetime"};
+};
+
+void test_construct_from_options() noexcept {
+  Options<tmpl::list<RegularSphWaveKerr>> opts("");
+  opts.parse(
+      "RegularSphWaveKerr:\n"
+      "  BlackHoleMass: 1.\n"
+      "  BlackHoleSpin: [0.1,0.2,0.3]\n"
+      "  BlackHoleCenter: [0.,0.,0.]\n"
+      "  WaveProfile:\n"
+      "    Gaussian:\n"
+      "      Amplitude: 1.\n"
+      "      Width: 1.\n"
+      "      Center: 0.");
+  CHECK(opts.get<RegularSphWaveKerr>() ==
+        CurvedScalarWave::AnalyticData::RegularSphericalWaveKerrSchild(
+            1.0, {{0.1, 0.2, 0.3}}, {{0.0, 0.0, 0.0}},
+            std::make_unique<MathFunctions::Gaussian>(1., 1., 0.)));
+}
+
+void test_serialize() noexcept {
+  CurvedScalarWave::AnalyticData::RegularSphericalWaveKerrSchild solution(
+      1.0, {{0.1, 0.2, 0.3}}, {{0.0, 0.0, 0.0}},
+      std::make_unique<MathFunctions::Gaussian>(1., 1., 0.));
+  Parallel::register_derived_classes_with_charm<MathFunction<1>>();
+  test_serialization(solution);
```

Comment:
Here too, pass the deserialized solution through a full test of all the values.

Reply:
Added such a test, please have a look and see if this is what you were thinking of..